src/wormc.nim

```diff
@@ -1,144 +1,165 @@
-import x11/[x, xlib, xutil]
-import std/[strutils, os]
-# import log
-import atoms
+import
+  std/[strutils, os],
+  x11/[x, xlib, xutil],
+  atoms
 
 converter toXBool(x: bool): XBool = x.XBool
 converter toBool(x: XBool): bool = x.bool
 
 type Layout = enum
   lyFloating, lyTiling
 
-proc main: void =
+proc formatMess(a: Atom, params: varargs[string, `$`]): array[5, clong] =
+  result[0] = a.clong
+
+  for i in 1 ..< 5:
+    if i < params.len:
+      result[i] = params[i].parseInt().clong
```

Comment:
Ok, I think I found an error. Theses indices are wrong. The `ith`-param should go into the `i+1` position of the result because the position 0 is for the atom. Sorry about this.

A correct version would be something like
```nim
for i in 0 ..< params.len:
  if i + 1 < result.len:
    result[i + 1] = params[i].parseInt().clong
```